R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,10 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  if (nzchar(master)) {
+    assign("spark.master", master, envir = sparkConfigMap)
```

Comment:
also I'd move this
```
if (nzchar(master)) {
     assign("spark.master", master, envir = sparkConfigMap)
```

to above L355, as
```
if (nzchar(master)) {
     sparkConfigMap[["spark.master"]] <- master
```

This is because of the user has explicitly set
`sparkR.session(spark.master = "yarn-client")` it should override whatever `master` is (that's in L361).